### PR TITLE
fix(elasticsearch sink): Fix data stream timestamp handling

### LINKED ISSUE #8267: Elasticsearch sink data stream mode test is failing
The test `sinks::elasticsearch::integration_tests::insert_events_in_data_stream` is currently failing on master due to a typo in that test. @jszwedko started [a branch](https://github.com/timberio/vector/tree/fix-elasticsearch-test) which fixes that typo. It also adds a missing `@timestamp` field, which results in the events being inserted without errors being displayed. This appears to have started with https://github.com/timberio/vector/commit/b6dc3b73bbdb7b2b9b8611ac098a4ec84a6a3f19 which added support for end-to-end acknowledgements to this sink, but that alone shouldn't have changed the inserted data format, so I am unclear why this has become necessary.

However, the search query still returns no events inserted. When querying the elasticsearch container, I can see 100 events inserted as expected. However, the `_index` metadata field appears to be incorrect, reading `.ds-logs-generic-default-2021.07.13-000001` instead of the random index generated in the test case. So it appears that the index is not being set correctly when the events are inserted in data stream mode.

### PR DESCRIPTION
**Pull request by bruceg**
Closes #8267 

The comparison here still isn't working right due to the additional timestamp and `data_stream` fields, and I don't think we should have to add a timestamp in the test to make this work. https://github.com/timberio/vector/blob/03a76433806a87899fa2f5e3325c071069f1f432/src/sinks/elasticsearch/mod.rs#L1513

### CODE REVIEW

**Review by jszwedko — APPROVED** [review]
This looks good, thanks!

File: src/sinks/elasticsearch/mod.rs [hunk 1]
```diff
@@ -36,6 +36,9 @@ use std::collections::{BTreeMap, HashMap};
 use std::convert::TryFrom;
 use vector_core::event::{Event, Value};
 
+/// The field name for the timestamp required by data stream mode
+const TIMESTAMP_KEY: &str = "@timestamp";
```
Comment (jszwedko) [inline src/sinks/elasticsearch/mod.rs:40]:
```suggestion
const DATASTREAM_TIMESTAMP_KEY: &str = "@timestamp";
```

Nit

**Comment by bruceg** [comment]
Started integration tests: https://github.com/timberio/vector/actions/runs/1031221885

File: src/sinks/elasticsearch/mod.rs [hunk 2]
```diff
@@ -180,14 +176,15 @@ impl DataStreamConfig {
         true
     }
 
-    fn update_timestamp(&self, mut event: Event) -> Event {
+    fn fixup_timestamp(&self, mut event: Event) -> Event {
```
Comment (jszwedko) [inline src/sinks/elasticsearch/mod.rs:179]:
```suggestion
    fn map_timestamp(&self, mut event: Event) -> Event {
```

Another nit.

**Review by jszwedko — APPROVED** [review]
Nice work, this looks good to me.

File: src/sinks/elasticsearch/mod.rs (hunk 1 above)
Comment (bruceg) [reply src/sinks/elasticsearch/mod.rs]: Done

File: src/sinks/elasticsearch/mod.rs (hunk 2 above)
Comment (bruceg) [reply src/sinks/elasticsearch/mod.rs]: Done

**Pull request merged**